Revit_Core_Engine/Compute/GenerateProfile.cs

```diff
@@ -0,0 +1,543 @@
+        {
+            List<BH.oM.Physical.Materials.IMaterialProperties> structuralProperties = material?.Properties?.Where(x => x is BH.oM.Structure.MaterialFragments.IMaterialFragment)?.ToList();
+            if (structuralProperties == null || structuralProperties.Count == 0)
+            {
+                BH.Engine.Base.Compute.RecordWarning($"Material for model behaviour of family {family.Name} could not be set because the source BHoM section property does not contain relevant material information. ElementId: {family.Id}");
+                family.SetParameter(BuiltInParameter.FAMILY_STRUCT_MATERIAL_TYPE, (int)Autodesk.Revit.DB.Structure.StructuralMaterialType.Other);
+                return false;
+            }
+            else if (structuralProperties.Select(x => x.GetType().FullName).Distinct().Count() > 1) 
+            {
+                BH.Engine.Base.Compute.RecordWarning($"Material for model behaviour of family {family.Name} could not be set because the source BHoM section property contains more than one relevant material information. ElementId: {family.Id}");
+                family.SetParameter(BuiltInParameter.FAMILY_STRUCT_MATERIAL_TYPE, (int)Autodesk.Revit.DB.Structure.StructuralMaterialType.Other);
+                return false;
+            }
+            
+            Type materialType = structuralProperties[0].GetType();
+            if (materialType == typeof(BH.oM.Structure.MaterialFragments.Steel))
+                return family.SetParameter(BuiltInParameter.FAMILY_STRUCT_MATERIAL_TYPE, (int)Autodesk.Revit.DB.Structure.StructuralMaterialType.Steel);
+            else if (materialType == typeof(BH.oM.Structure.MaterialFragments.Concrete))
+                return family.SetParameter(BuiltInParameter.FAMILY_STRUCT_MATERIAL_TYPE, (int)Autodesk.Revit.DB.Structure.StructuralMaterialType.Concrete);
+            else if (materialType == typeof(BH.oM.Structure.MaterialFragments.Timber))
+                return family.SetParameter(BuiltInParameter.FAMILY_STRUCT_MATERIAL_TYPE, (int)Autodesk.Revit.DB.Structure.StructuralMaterialType.Wood);
+            else
+            {
+                BH.Engine.Base.Compute.RecordWarning($"Material for model behaviour of family {family.Name} has been set to \'Other\' set because the material of source BHoM section property is not supported by Revit in this context. ElementId: {family.Id}");
+                return family.SetParameter(BuiltInParameter.FAMILY_STRUCT_MATERIAL_TYPE, (int)Autodesk.Revit.DB.Structure.StructuralMaterialType.Other);
+            }
+        }
+
+        /***************************************************/
+
+        private static void ICopyDimensions(this BH.oM.Spatial.ShapeProfiles.IProfile sourceProfile, FamilySymbol targetSymbol, RevitSettings settings = null)
+        {
+            CopyDimensions(sourceProfile as dynamic, targetSymbol, settings);
+        }
```

Comment:
Should this and all related `CopyDimensions` methods be moved to a separate file?

Reply:
Noo, I would rather keep it as a private method because it is only relevant in the context of the public method at the top of the file (`GenerateProfile`) - personally I cannot imagine calling it anywhere else.

However, if you see any usage of it in other contexts, we can think of making it public.